include/fir/Type.h

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FIR_TYPE_H
+#define FIR_TYPE_H
+
+#include "mlir/IR/Types.h"
+#include <variant>
+
+namespace llvm {
+class StringRef;
+template<typename> class ArrayRef;
+class hash_code;
+}
+
+namespace fir {
```

Comment:
My understanding is that the f18->mlir bridge will be in namespace fortran, so there's no notational convenience to using Fortran::fir in the bridge.  I would choose Fortran::fir if code in namespace fir will be referencing things in namespace Fortran, such as symbols, symbol tables or parse tree nodes because you can't have a fir without a Fortran so they might as well be combined.

Reply:
Re: Steve's comment. That is correct.